### PR TITLE
fix: handle manually pinned command

### PR DESCRIPTION
**Pull request by estrada9166**
- Closes  [AW-57](https://cypress-io.atlassian.net/browse/AW-57)

### Additional details

---

> [!NOTE]
> **Medium Risk**
> Changes tap pin reconciliation and AUT DOM restore when pins come from mixed tap/reporter sources; behavior is security-adjacent for agents reading DOM state but is heavily covered by new e2e and unit tests.
> 
> **Overview**
> **Tap now treats reporter command pins as first-class:** `run-state` exposes a `pinned` block when someone pins via the command log (not only via `cypress tap pin`), and `pin --clear` can release that pin through the app’s unpin path.
> 
> The binding reads the **authoritative** pin from the runner snapshot store (`getPinnedSnapshot`), merges it with tap’s own captured live DOM when tap had pinned first, and reconciles cases where the UI replaces tap’s pin without an unpin event. Snapshot identity uses string log ids plus **`testId`** on snapshot props (driver + `AutSnapshot`) so pins resolve to the correct test/attempt via new **`attemptOfLog`** logic.
> 
> Coverage adds a full e2e flow for reporter click-pin / snapshot toggle / clear, plus unit tests for UI pins, stale eviction, and DOM restore edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit adf095206f56113b64fdfc12be7331cdf75b8af1. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### PR Tasks

- [ ] Is there an associated issue with maintainer approval for PR submission? 
- [ ] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

[AW-57]: https://cypress-io.atlassian.net/browse/AW-57?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/app/src/runner/iframe-model.ts [hunk 1]
```diff
@@ -3,7 +3,10 @@ import { useAutStore } from '../store'
 import type { EventManager } from './event-manager'
 
 export interface AutSnapshot {
-  id?: number
+  /** The driver's id for the log these snapshots belong to, e.g. `log-http://localhost:3000-4`. */
+  id?: string
+  /** Id of the test that log belongs to — a log id resolves only within its test. */
+  testId?: string
```
Comment (davidr-cy) [inline packages/app/src/runner/iframe-model.ts]: remove comments

File: packages/app/src/tap/commands/pin.ts [hunk 2]
```diff
@@ -56,33 +74,89 @@ const onExternalUnpin = (): void => {
...
+// What the AUT is showing, whoever pinned it. A reporter click on another command
+// replaces tap's pin with no unpin event to hear, so the command showing then
+// comes from the app, while the captured DOM — tap's alone to put back — stays
+// with it.
+const currentPin = (): CurrentPin | undefined => {
+  const showing = uiPin()
+
+  if (!pinned) {
+    return showing
+  }
```
Comment (davidr-cy) [inline packages/app/src/tap/commands/pin.ts]: looking at this now what does the local pinned get us that `uiPin` doesn't? IE: do we need this dual state? Is it an issue that the way we pin snapshots via tap doesn't reflect in the state that `uiPin` pulls?

Comment (estrada9166) [reply packages/app/src/tap/commands/pin.ts]: - `uiPin()` tells us what snapshot the AUT is currently showing, regardless of whether it was pinned by tap or the reporter.
- Local pinned tells us what tap owns:
  - the original DOM tap must restore;
  - snapshot identity for stale-run/eviction detection;
  - command identity for in-place moves;
  - unpin-listener lifecycle.
  
Tap pins already update the store read by uiPin(). Therefore:
- We need local ownership metadata.
- We do not need local state as a second source of truth for the visible pin.
- uiPin() should determine what is displayed.
- Local state should only supplement it with restoration/lifecycle data.

I have updated this logic

Comment (davidr-cy) [reply packages/app/src/tap/commands/pin.ts]: I was actually just about to push a stacked branch with some changes using what you did that removes all the local pinned state.

**Review by davidr-cy — APPROVED** [review]

**Review by davidr-cy — APPROVED** [review]

File: packages/driver/src/cypress/log.ts [hunk 3]
```diff
@@ -15,7 +15,9 @@ import type { StateFunc } from './state'
 // including the intermediate $Log interface
 const groupsOrTableRe = /^(groups|table)$/
 const parentOrChildRe = /parent|child|system/
-const SNAPSHOT_PROPS = 'id snapshots $el url coords highlightAttr scrollBy viewportWidth viewportHeight'.split(' ')
+// `testId` rides along because every per-log lookup is keyed by (testId, logId);
+// a log id alone cannot be resolved back to a test.
```
Comment (davidr-cy) [inline packages/driver/src/cypress/log.ts]: remove thi
```suggestion
```
